avocado/plugins/runner_nrunner.py

```diff
@@ -233,29 +232,29 @@ def _get_all_runtime_tasks(test_suite):
         runtime_tasks = []
         test_result_total = test_suite.variants.get_number_of_tests(test_suite.tests)
         no_digits = len(str(test_result_total))
+        # define execution order
         execution_order = test_suite.config.get('run.execution_order')
         if execution_order == "variants-per-test":
-            for index, (runnable, variant) in enumerate(((test, variant)
-                                                         for test in test_suite.tests
-                                                         for variant in test_suite.variants.itertests()),
-                                                        start=1):
-                runtime_tasks.extend(Runner._create_runtime_tasks_for_test(
-                    test_suite,
-                    runnable,
-                    no_digits,
-                    index,
-                    variant))
+            test_variant = [(test, variant) for test in test_suite.tests
+                            for variant in test_suite.variants.itertests()]
         elif execution_order == "tests-per-variant":
-            for index, (runnable, variant) in enumerate(((test, variant)
-                                                         for variant in test_suite.variants.itertests()
-                                                         for test in test_suite.tests),
-                                                        start=1):
-                runtime_tasks.extend(Runner._create_runtime_tasks_for_test(
-                    test_suite,
-                    runnable,
-                    no_digits,
-                    index,
-                    variant))
+            test_variant = [(test, variant)
+                            for variant in test_suite.variants.itertests()
+                            for test in test_suite.tests]
+        # decide if a copy of the runnable is needed, in case of more variants
+        # than tests
+        copy_runnable = False
```

Comment:
I'm not a fan of assignments that can be made irrelevant because of a following conditional that will always run.  My suggestion here is:

```python
copy_runnable = len(test_variant) > len(test_suite.tests)
```

Reply:
OKay, either one works for me. I'll change that.